src/lib.rs

```diff
@@ -572,7 +574,15 @@ impl<C: Context> Secp256k1<C> {
 
     /// Returns the required memory for a preallocated context buffer in a generic manner(sign/verify/all)
     pub fn preallocate_size_gen() -> usize {
-        unsafe { ffi::secp256k1_context_preallocated_size(C::FLAGS) }
+        assert!(mem::align_of::<AlignType>() >= mem::align_of::<u8>());
+        assert!(mem::align_of::<AlignType>() >= mem::align_of::<usize>());
+        assert!(mem::size_of::<AlignType>() >= mem::size_of::<usize>());
+        assert_eq!(mem::size_of::<AlignType>(), mem::size_of::<&AlignType>());
```

Comment:
This last assertion will be untrue if `AlignType` is not `usize`.

Reply:
your'e right. maybe `assert!(mem::align_of::<AlignType>() => mem::align_of::<&AlignType>());`? or it's already too much assertions?

Reply:
Heh, you can do the `=>`. I think it's fine to have all these assertions.